docs/DeveloperGuide.md

```diff
@@ -1,34 +1,426 @@
+the CommandStack and pass it to the Command class to execute the undo action by removing it from the module
+contained in the hashmap. Also, this note will be added to the redoStack list.
+<br><br>
+If operation is to redo removed notes, the CommandStack will remove the last added note in redoStack list and
+pass it to the Command class to execute the redo action by adding this note into the module contained in the
+hashmap.
+<br><br>
+The reason why we chose two linked lists to support these operations is because it reduces the SLOC needed to
+write the logic. An alternative is to actually remember the state of the hashmap before an operation and save
+it to another hashmap. However, this approach will take up more memory and reduces the performance of the application. 
+
+[comment]: # (@@author )
+## 4. Testing 
+
+## Appendix A: Product Scope 
+### Target User Profile  
+
+Mainly NUS students and professors who :
+* need to manage their tasks, events and deadlines
+* prefer desktop app over other types
+* are new to the faculty (or freshmen) , and require knowledge on conducive areas to study
+* have specific needs for a conducive Study Area
+
+### Value proposition    
+
+[comment]: # (to be finalised)  
+
+You can efficiently manage tasks, in terms of priority. You can also efficiently find a study area that
+meets your needs and is conducive, should you urgently need one.
+
+## Appendix B: User Stories    
+ 
+ |Version| As a ... | I want to ... | So that I can ...| 
+ |--------|----------|---------------|------------------|
+ |v1.0|new user|see list of commands supported|refer to them when I forget how to use the application|  
+ |v1.0|user|add a new task|keep track of it|  
+ |v1.0|user|edit a task|change it's details if required| 
+ |v1.0|user|delete a task|remove tasks that are cancelled| 
+ |v1.0|user|view my list of tasks sorted based on their date|be aware of all the tasks that are due/happening soon|  
+ |v1.0|user|view my list of tasks sorted based on their priority|be aware of the more important tasks|  
+ |v1.0|user|find a task by name|locate a task without having to go through the entire list|    
+ |v1.0|student|find a Study Area based on location and environment|have a conducive space to study|
+ |v1.0|professor|locate a Study Area that is outdoors| conduct consultations without worrying of making noise|
+ |v2.0|user|mark deadline as done|check to see if I have pending deadlines|
+ |v2.0|user|to delete a Module|clean up my finished notes|
+ |v2.0|user|create a Module|add notes inside|
+ |v2.0|user|undo an added note|increase my efficiency|
+ |v2.0|user|redo a removed notes|increase my efficiency|
+ |v2.0|user|create a schedule based on requirements|customise my tasks accordingly|
+
+## Appendix C: Non-Functional Requirements    
+  {Give non-functional requirements}    
+    
+[comment]: # (@@author NizarMohd)
+## Appendix D: Glossary    
```

Comment:
Very detailed glossary, it might be easier to reference if it was used in a table instead, what do you think?